src/Storage/ClaimController.ts

```diff
@@ -178,10 +210,37 @@ export class ClaimController {
   }
 
   private downloadEntryClaim = async ({ entry, ...rest }: { entry: Entry }) => {
+    const { ipfsHash } = entry
+    const downloadClaim = async () => {
+      try {
+        return await this.ipfs.cat(ipfsHash)
+      } catch (error) {
+        if (isFetchTimeoutError(error)) throw new IPFSTimeoutError(ipfsHash)
+        else throw new IPFSGenericError(ipfsHash, error)
+      }
+    }
+    const parseClaim = (serialized: string) => {
+      try {
+        const parsedJson = JSON.parse(serialized)
+        return {
+          ...parsedJson,
+          dateCreated: new Date(parsedJson.dateCreated),
+        }
+      } catch (error) {
+        throw new InvalidClaim(ipfsHash, FailureReason.InvalidJson)
+      }
+    }
     const logger = this.logger.child({ method: 'downloadEntryClaim' })
-    logger.trace('starting claim download')
-    const claim = await this.downloadClaim(entry.ipfsHash)
-    logger.trace('finished claim download', claim)
+
+    logger.trace({ ipfsHash }, 'Starting claim download')
+
+    const serialized = await downloadClaim()
```

Comment:
We should probably pass in the hash to downloadClaim.

Reply:
See comment above.